x-pack/plugins/observability_solution/synthetics/server/alert_rules/status_rule/query_monitor_status_alert.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+import times from 'lodash/times';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { cloneDeep, intersection } from 'lodash';
+import { createEsParams, SyntheticsEsClient } from '../../lib';
+import {
+  OverviewPendingStatusMetaData,
+  OverviewPing,
+  OverviewStatus,
+  OverviewStatusMetaData,
+} from '../../../common/runtime_types';
+import { FINAL_SUMMARY_FILTER } from '../../../common/constants/client_defaults';
+
+const DEFAULT_MAX_ES_BUCKET_SIZE = 10000;
+
+const fields = [
+  '@timestamp',
+  'summary',
+  'monitor',
+  'observer',
+  'config_id',
+  'error',
+  'agent',
+  'url',
+  'state',
+  'tags',
+];
+
+export async function queryMonitorStatusForAlert(
+  esClient: SyntheticsEsClient,
+  monitorLocationIds: string[],
+  range: { from: string; to: string },
+  monitorQueryIds: string[],
+  monitorLocationsMap: Record<string, string[]>,
+  monitorQueryIdToConfigIdMap: Record<string, string>
+): Promise<
+  Omit<
+    OverviewStatus,
+    | 'disabledCount'
+    | 'allMonitorsCount'
+    | 'disabledMonitorsCount'
+    | 'projectMonitorsCount'
+    | 'disabledMonitorQueryIds'
+    | 'allIds'
+  >
+> {
+  const idSize = Math.trunc(DEFAULT_MAX_ES_BUCKET_SIZE / monitorLocationIds.length || 1);
+  const pageCount = Math.ceil(monitorQueryIds.length / idSize);
+  let up = 0;
+  let down = 0;
+  const upConfigs: Record<string, OverviewStatusMetaData> = {};
+  const downConfigs: Record<string, OverviewStatusMetaData> = {};
+  const monitorsWithoutData = new Map(Object.entries(cloneDeep(monitorLocationsMap)));
+  const pendingConfigs: Record<string, OverviewPendingStatusMetaData> = {};
+
+  await pMap(
```

Comment:
Is there a reason we can't use `msearch` here as well instead of `pmap`? The goal just to split this out and not re-write it at all?